src/python/include/Simplex_tree_interface.h

```diff
@@ -163,10 +164,11 @@ class Simplex_tree_interface : public Simplex_tree<SimplexTreeOptions> {
     using Filtered_edge = std::tuple<Vertex_handle, Vertex_handle, Filtration_value>;
     std::vector<Filtered_edge> edges;
     for (Simplex_handle sh : Base::skeleton_simplex_range(1)) {
-      if (Base::dimension(sh) == 1) {
-        typename Base::Simplex_vertex_range rg = Base::simplex_vertex_range(sh);
-        std::vector<Vertex_handle> rips_edge(rg.begin(), rg.end());
-        edges.push_back(std::make_tuple(rips_edge[0], rips_edge[1], Base::filtration(sh)));
+      typename Base::Simplex_vertex_range rg = Base::simplex_vertex_range(sh);
+      auto rg_begin = rg.begin();
+      // We take only edges into account
+      if (std::distance(rg_begin, rg.end()) == 2) {
+        edges.emplace_back(*rg_begin, *std::next(rg_begin), Base::filtration(sh));
```

Comment:
This still computes ++ multiple times. Computing things once could look like:
```c++
auto&& rg = Base::simplex_vertex_range(sh);
auto vit = std::begin(rg);
auto e = std::end(rg);
assert(vit!=e);
Vertex_handle v = *vit;
if(++vit==e) continue; // vertex
Vertex_handle w = *vit;
assert(++vit==e);
edges.emplace_back(v, w, Base::filtration(sh));
```
But we should probably go back to testing `Base::dimension(sh) == 1`, which will be easy to change once #183 is implemented.

Reply:
Ok, I got it, it is because of std::distance implementation.
If you are ok with it, I rolled it back with `Base::dimension(sh) == 1` to make it easier to understand: c3a4fc7b.
This is a perfect example where #183 would be helpful.